scripts/descriptors/flotsamgenerator.lua

```diff
@@ -0,0 +1,30 @@
+local spawn_task
+if TheWorld.components.flotsamgenerator and TheWorld.components.flotsamgenerator.ScheduleGuaranteedSpawn then
+	spawn_task = util.getupvalue(TheWorld.components.flotsamgenerator.ScheduleGuaranteedSpawn, "_guaranteed_spawn_tasks")
+end
+
+local function Describe(self, context)
+	local descriptions = {}
+	if spawn_task then
```

Comment:
As a general rule, early returns are better than nested logic.

So this would need to be something like
```lua
if not spawn_task then
return
end
...
```